query-algebrizer/src/validate.rs

```diff
@@ -229,4 +274,177 @@ mod tests {
             _ => panic!(),
         };
     }
+
+
+    /// Tests that the top-level form is a valid `not`, returning the clauses.
+    fn valid_not_join(parsed: FindQuery, expected_unify: UnifyVars) -> Vec<WhereNotClause> {
+        // filter all the not clauses
+        let mut nots = parsed.where_clauses.iter().filter(|&x| match *x {
+            WhereClause::NotJoin(_) => true,
+            _ => false,
+        });
+
+        // there should be only one not clause
+        let clause = nots.next().unwrap().clone();
+        assert_eq!(None, nots.next());
```

Comment:
One advantage of `assert_eq!` is that it will show you the `Some(...)` in the failure case.  For this reason, I prefer asserting equality.